### DISCUSSION TITLE
Smart search ... no good matches

### DISCUSSION THREAD

**Discussion by emigrating** [immich-app/immich#13285] [Q&A] [answered]
I believe there is something wrong with my install, settings or, erm, brain :)

Have the instance installed on a TrueNAS Scale machine, seem to be working fine. Installed the ML container on a different host with a beefy GPU. Also seems to work fine. Faces are detected and I see the GPU under load when I start the ML based jobs.

Great. I then go to search for something, anything. Say a "Pizza" and it shos me photos of SF cityscapes, a baby holding a birthday present, a yellow digger and a red crane.. Not a single pizza in sight. (Yes, I do have pizzas in the test library). Oh well, I search for "boat" and it shows me street-performers in Central Park, a family gathering, an Xbox redeemable code and a cake.

Hmm, something ain't right. So I dive into the clip settings and start looking at different models etc, try a handful of different ones. Have to rerun the smart search job after each change, see my GPU chugging along and always I get the same results. The actual photos displayed seem to be different for the same search terms for each clip, but they stay the same for each search term on the same clip so to speak.

Does anyone have some insight at all here? I get no errors on the remote ML container nor do I get any errors on the TrueNAS install. Everything *seems* to be working but the search results I get seem completely random. Unless I click an actual person or three, then it shows everything from the right person or group of people (and yes, I know this is a different model but it shows that the remote ML container is actually working).

**Comment by alextran1502** [comment]
Which CLIP model are you using? Can you try describe the photo with the pizza?

**Reply by emigrating** [reply]
I have tried pretty much all of them..

For right now I'm on **ViT-L-16-SigLIP-384__webli**, but the results have been the same on all the different ones. It just gives me, IMHO, unrelated matches.

Here are some of the pizza photos;

![image](https://github.com/user-attachments/assets/9e20d333-cdd0-4458-8124-fa7d556b5783)
![image](https://github.com/user-attachments/assets/1f932afb-6035-45cc-85a2-2676105b5274)
![image](https://github.com/user-attachments/assets/7ce56e7d-15d9-4b92-8c3f-528657b49f9c)

and the results I get are;

![image](https://github.com/user-attachments/assets/820eb714-a261-49a0-97b1-f05c6879c74b)

**Comment by mertalev** [comment]
Are you using OpenVINO? There have been a few reported issues with the latest release, one of which is that the results for smart search are worse than on CPU.

**Reply by emigrating** [reply]
Yes, I am and I did see some talk about that, but I would call the results I get "complete and utter horses##t" rather than worse 😄

I mean, technically they *are* worse, but...

**Reply by mertalev** [reply]
You can try using the 1.118 image for the machine learning service and re-run smart search on all assets.

**Reply by emigrating** [reply]
> You can try using the 1.118 image for the machine learning service and re-run smart search on all assets.

Latest seems to be 1.117? But I pulled the latest pr- release and retried. Same exact thing, the detection goes through fine but same crappy results.

~~Gonna try a from-scratch install of the docker containers on another LXC now to figure out if it's the fact I'm running on TrueNAS or not.~~

Hmm, so I installed the docker containers (bar the ML one - still running the later PR rev) on another host (ie not the TrueNAS). Seems to have grabbed the same version (1.117). Set this up with the same TestPhotoFolder as TN. Configured it to use the same ML container. And this works.

Only difference seems to be that this immich instance is using the default openai CLIP. Gonna change it over and see what happens.

**Reply by mertalev** [reply, marked as answer]
Oh, my bad: the last unaffected version is 1.116.2. Not sure why my mind went to 1.118 haha.

**Reply by emigrating** [reply]
> Oh, my bad: the last unaffected version is 1.116.2. Not sure why my mind went to 1.118 haha.

Pulling the 1.116.2 container and everything works. I *think* the search results are still slighty better using ViT-B-32__openai, but thats an acceptable tradeoff as the others seem to recognize "more" things (more of each if you like).

So, thanks. Downgrading the ML container worked wonders.

**Reply by a-jackson** [reply]
I was just having the same issue. Whatever I search, I get the same nonsense results. I'm also on OpenVino. Downgraded to 1.116.2 and reran smart search job but no difference. Is anything else needed other than swapping the image?

**Reply by mertalev** [reply]
> I was just having the same issue. Whatever I search, I get the same nonsense results. I'm also on OpenVino. Downgraded to 1.116.2 and reran smart search job but no difference. Is anything else needed other than swapping the image?

Try deleting your model-cache volume and re-running smart search on all assets.

**Reply by a-jackson** [reply]
Thanks that worked
